lib/bolt/transport/simple.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'logging'
+require 'bolt/result'
+require 'bolt/shell'
+require 'bolt/transport/base'
+
+module Bolt
+  module Transport
+    # A simple transport has a single connection per target and delegates its
+    # operation to a target-specific shell.
+    class Simple < Base
+      def with_connection(_target)
```

Comment:
I wonder if this can also be generalized to
```
conn = Connection.new(target, @transport_logger)
        conn.connect
        yield conn
      ensure
        begin
          conn&.disconnect
        rescue StandardError => e
          logger.info("Failed to close connection to #{target.safe_name} : #{e.message}")
        end
      end
```
with connect and disconnect stubbed for transports that don't use it. That might be more effort than it's worth.

Reply:
I definitely considered that and I think it probably makes sense, but it's another one where I wanted to wait and see how the other transports turn out.